packages/nextjs/hooks/scaffold-eth/useDeployedContractInfo.ts

```diff
@@ -30,16 +30,22 @@ export const useDeployedContractInfo = <TContractName extends ContractName>(cont
       // If contract code is `0x` => no contract deployed on that address
       if (code === "0x") {
         setStatus(ContractCodeStatus.NOT_FOUND);
-        return;
+      } else {
+        setStatus(ContractCodeStatus.DEPLOYED);
       }
-      setStatus(ContractCodeStatus.DEPLOYED);
     };
 
     checkContractDeployment();
   }, [isMounted, contractName, deployedContract, provider]);
 
+  let isLoading = status === ContractCodeStatus.LOADING;
+
+  if (status === ContractCodeStatus.NOT_FOUND) {
+    isLoading = false;
+  }
+
   return {
     data: status === ContractCodeStatus.DEPLOYED ? deployedContract : undefined,
-    isLoading: status === ContractCodeStatus.LOADING,
+    isLoading: isLoading,
   };
```

Comment:
This change can be reverted, I believe. If status equals `NOT_FOUND`, then `isLoading` would already be `false`, so no need for this additional state check. Setting `isLoading` in the returned object directly is fine.

Reply:
Reverted!